packages/components/button/src/button.tsx

```diff
@@ -36,7 +39,11 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
     },
     ref
   ): React.ReactElement => (
-    <StyledButton onClick={onClick} ref={ref}>
+    <StyledButton
+      data-testid={testId || kebabCase(`button-${children}`)}
```

Comment:
This doesn't feel safe, as `children` supports a `ReactNode` as opposed to just a string. If something other than a string were to be passed in I'm not quite sure what would happen here.

Reply:
Got it, is there anyway to derive the button name? Or is it just easiest to pass in a custom `data-testid`